source/braille.py

```diff
@@ -1444,11 +1445,141 @@ def _setCursor(self, info: textInfos.TextInfo):
 class ReviewTextInfoRegion(TextInfoRegion):
 
 	allowPageTurns=False
+	_currentSelection: textInfos.TextInfo | None = None
+	_currentScript: _ScriptFunctionT | None
+	_readingUnitContainsSelectedCharacters: bool
+
+	def _getSelection(self) -> textInfos.TextInfo:
+		"""Gets selection for use in update function.
+		:return: reading unit, when it contains selected characters,
+		or review position, when reading unit is outside of selection,
+		there is no selection, or showing selection is disabled
+		"""
+		self._readingUnitContainsSelectedCharacters = False
+		if not config.conf["braille"]["showSelection"]:
+			self._currentSelection = None
+			return api.getReviewPosition().copy()
+		info: textInfos.TextInfo
+		try:
+			info = self.obj.makeTextInfo(textInfos.POSITION_SELECTION)
+		except (LookupError, RuntimeError):
+			self._currentSelection = None
+			return self._collapsedReviewPosition()
+		if info.isCollapsed:
+			# Cursor
+			self._currentSelection = None
+			return self._collapsedReviewPosition()
+		# Selection
+		if self._currentSelection is None or self._currentSelection != info:
+			# Selection changed, update also review position
+			self._currentSelection = info.copy()
+			if self.obj.isTextSelectionAnchoredAtStart:
+				self._currentSelection.move(textInfos.UNIT_CHARACTER, -1, "end")
+			self._currentSelection.collapse(end=self.obj.isTextSelectionAnchoredAtStart)
+			# Enqueue to avoid recursion error
+			queueHandler.queueFunction(queueHandler.eventQueue, api.setReviewPosition, self._currentSelection)
+			self._currentSelection = info.copy()

```

Comment:
I think this is an unnecessary duplicate of line 1475. If not, please explain in code why you're setting it again here.